test/test_collection.py

```diff
@@ -2253,6 +2253,29 @@ def afind(*args, **kwargs):
         for helper, args in helpers:
             helper(*args, let={})  # type: ignore
 
+    def assertRepr(self, obj):
+        new_obj = eval(repr(obj))
+        self.assertEqual(type(new_obj), type(obj))
+        self.assertEqual(repr(new_obj), repr(obj))
+
+    def test_index_model_repr(self):
+        self.assertRepr(IndexModel("hello"))
```

Comment:
We should also have at least one assert with a literal string, eg:
```"IndexModel({'hello': 1, 'world': -1}, name='hello_world')"```

That way we'll know if the format changes by accident.

Reply:
https://github.com/mongodb/mongo-python-driver/pull/1909/commits/f4b255b42684e2436a4d9db0a7482521228f45e8#diff-97ad6fe624c2aee90c37ee01ad11452de5c87847a24567b0cab10c41824bf22fR2282

Reply:
We need to compare the string literal value:

`self.assertEqual(repr(IndexModel(...)), "IndexModel({'hello': 1, 'world': -1}, name='hello_world')")`

Reply:
https://github.com/mongodb/mongo-python-driver/pull/1909/commits/09ef323e6c98d862e13a5f4a8331051880b2b380#diff-97ad6fe624c2aee90c37ee01ad11452de5c87847a24567b0cab10c41824bf22fR2282-R2283